tests/test_machine.py

```diff
@@ -39,16 +39,18 @@ def test_infinite_command(self) -> None:
 
 
 def test_before_boot() -> None:
-    m = Machine()
-    if m.args.verb == Verb.shell:
-        pytest.skip("Shell never boots the machine.")
-    with pytest.raises(AssertionError):
-        m.run(["ls"])
+    with pytest_skip_not_supported():
+        m = Machine()
+        if m.args.verb == Verb.shell:
+            pytest.skip("Shell never boots the machine.")
+        with pytest.raises(AssertionError):
+            m.run(["ls"])
 
 
 def test_after_shutdown() -> None:
-    with Machine() as m:
+    with pytest_skip_not_supported(), Machine() as m:
```

Comment:
I'd put `Machine() as m` in a nested with block so it's more clear that the first context manager also applies to it.